vkcs/provider.go

```diff
@@ -166,50 +186,42 @@ func Provider() *schema.Provider {
 			"auth_url": {
 				Type:        schema.TypeString,
 				Optional:    true,
-				DefaultFunc: schema.EnvDefaultFunc("OS_AUTH_URL", defaultIdentityEndpoint),
 				Description: "The Identity authentication URL.",
 			},
 			"project_id": {
 				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("OS_PROJECT_ID", ""),
+				Optional:    true,
 				Description: "The ID of Project to login with.",
 			},
 			"password": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
```

Comment:
Some fields became optional since I removed the defaults to match the scheme of the new provider. We set defaults and validate fields at the provider configuration stage.

https://github.com/hashicorp/terraform-plugin-framework/issues/539